src/lsp/sql_parser/lexer.mll

```diff
@@ -0,0 +1,88 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                        SuperBOL OSS Studio                             *)
+(*                                                                        *)
+(*  Copyright (c) 2022-2023 OCamlPro SAS                                  *)
+(*                                                                        *)
+(* All rights reserved.                                                   *)
+(* This source code is licensed under the GNU Affero General Public       *)
+(* License version 3 found in the LICENSE.md file in the root directory   *)
+(* of this source tree.                                                   *)
+(*                                                                        *)
+(**************************************************************************)
+
+{
+  open Grammar
+
+  (* Compléter la fonction de traduction selon les tokens *)
+  (*In alphabetical order*)
+  let get_keyword = function
+   | "AS" -> AS
+   | "AT" -> AT
+   | "BEGIN" -> BEGIN
+   | "BY" -> BY
+   | "COMMIT" -> COMMIT
+   | "CONNECT" -> CONNECT
+   | "DECLARE" -> DECLARE
+   | "DISCONNECT" -> DISCONNECT
+   | "DSN" -> DSN
+   | "END" -> END
+   | "EXEC" -> EXEC
+   | "FOUND" -> FOUND
+   | "GOTO" -> GOTO
+   | "IDENTIFIED" -> IDENTIFIED
+   | "INCLUDE" -> INCLUDE
+   | "NO" -> NO
+   | "NOT" -> NOT
+   | "PERFORM" -> PERFORM
+   | "PROMPT" -> PROMPT
+   | "RELEASE" -> RELEASE
+   | "RESET" -> RESET
+   | "RETURNING" -> RETURNING
+   | "ROLLBACK" -> ROLLBACK
+   | "SAVEPOINT" -> SAVEPOINT
+   | "SECTION" -> SECTION
+   | "START" -> START
+   | "SQL" -> SQL
+   | "SQLCA" -> SQLCA
+   | "SQLDA" -> SQLDA
+   | "SQLERROR" -> SQLERROR
+   | "SQLWARNING" -> SQLWARNING
+   | "TO" -> TO
+   | "TRAN" -> TRANSACTION
+   | "TRANSACTION" -> TRANSACTION
+   | "USER" -> USER
+   | "USING" -> USING
+   | "WHENEVER" -> WHENEVER
+   | "WITH" -> WITH
+   | "WORK" -> WORK
+   | s -> WORD s (* mot non reconnu : probablement du SQL "brut", on le garde tel quel *)
```

Comment:
Thinking about it, let's just name this "TOKEN" instead of "WORD": this sounds more generic.